[nucleus-3.0] cta component (#214)
centrica-engineering/nucleus-docs: docs/components/ns-cta.md

### Differences between `direct` and `text` types

> The `direct` amd `text` types of `ns-cta` are there to offer hierarchical structure to the page.

Reviewer: Typo.

```suggestion
The `direct` and `text` types of `ns-cta` are there to offer hierarchical structure to the page.
```

If there is a group of more than three actions it would be preferable to make all these CTAs the `text` type. Or to find the main action in this group and place it in an alternative, more suitable component such as `ns-lockup` or `ns-standout` to establish hierarchy.

### When to use `submit` and `reset` types

> The `submit` and `reset` types of `ns-cta` are there to facilitate the form component to do submit and reset actions respectively.

Reviewer: Could we use `ns-form` instead of 'the form component'?

```suggestion

The `submit` and `reset` types of `ns-cta` are there to facilitate `ns-form` to do submit and reset actions respectively.
```

> The `submit` type provides native-like submit action. It can be used in the page with form to do submit action. Upon submission which automatically validates the form elements and reports warning / submits the form.

Reviewer: ```suggestion
The `submit` type is to be used in a page with `ns-form` to perform a native-like `submit` action. Upon submission it automatically validates the elements in `ns-form` and reports any warnings or will submit `ns-form`.
```

> The `reset` type provides native-like reset action. When wrapped in form component, `reset` type triggers reset action on all form element components enclosed within the form and reset them back to their default / initial value. 

Reviewer: ```suggestion
The `reset` type is to be used in a page with `ns-form` to perform a native-like `reset` action. It will reset the elements back to their default or initial value.
```
